src/pages/payments/refund/Refund.tsx

```diff
@@ -0,0 +1,213 @@
+    invoices.map((invoiceId: string) => {
+      const invoiceItem = payment?.data.data.invoices.find(
+        (invoice: Invoice) => invoice.id == invoiceId
+      );
+      if (invoiceItem)
+        formik.setFieldValue('invoices', [
+          ...formik.values.invoices,
+          {
+            amount:
+              invoiceItem?.paid_to_date >
+              payment?.data.data.amount - payment?.data.data.refunded
+                ? payment?.data.data.amount - payment?.data.data.refunded
+                : invoiceItem?.paid_to_date,
+            invoice_id: invoiceItem?.id,
+            credit_id: '',
+            id: '',
+          },
+        ]);
+    });
+  }, [invoices]);
+
+  useEffect(() => {
+    let total = 0;
+    formik.values.invoices.map((invoice: any) => {
+      total = total + Number(invoice.amount);
+      setInvoices(
+        invoices.filter((invoiceId: string) => invoiceId != invoice.invoice_id)
+      );
+    });
+  }, [formik.values.invoices]);
+
+  return (
+    <Card
+      title={t('apply_payment')}
+      disableSubmitButton={formik.isSubmitting}
+      onFormSubmit={formik.handleSubmit}
+      withSaveButton
+    >
+      <Element leftSide={t('number')}>
+        <InputField disabled value={payment?.data.data.number} />
+      </Element>
+      <Element leftSide={t('amount')}>
+        <InputField
+          disabled
+          value={payment?.data.data.amount - payment?.data.data.refunded}
+        />
+      </Element>
+      <Element leftSide={t('applied')}>
+        <InputField disabled value={payment?.data.data.applied} />
+      </Element>
+      <Element leftSide={t('date')}>
+        <InputField
+          type="date"
+          value={formik.values.date}
+          onChange={formik.handleChange}
+        />
+      </Element>
+      <Element leftSide={t('invoices')}>
+        <SelectField
+          value=""
+          onChange={(event: any) => {
+            if (
+              formik.values.invoices.filter(
+                (invoices: any) => invoices.invoice_id == event.target.value
+              ).length < 1
+            )
+              setInvoices([...invoices, event.target.value]);
+          }}
+        >
+          <option value="" disabled></option>
+          {payment?.data.data.invoices &&
+            payment?.data.data.invoices.map(
+              (invoice: Invoice, index: number) => {
+                return (
+                  <option key={index} value={invoice.id}>
+                    {invoice.number}
+                  </option>
+                );
+              }
+            )}
+        </SelectField>
+        {errors?.errors.invoices && (
+          <Alert type="danger">{errors.errors.invoices}</Alert>
+        )}
+      </Element>
+      {payment?.data.data &&
+        formik.values.invoices.map((invoiceitem: any, index: number) => {
+          const invoiceItem = payment?.data.data.invoices.find(
+            (invoice: Invoice) => invoice.id == invoiceitem.invoice_id
+          );
+
```

Comment:
Can you explain what this is doing?

Reply:
Since request only holds invoice id and amount i need to search for invoice by id in order to access other data like paid_to_date i need to calculate final amount.